mmstudio/src/main/java/org/micromanager/acquisition/internal/acqengjcompat/AcqEngJAdapter.java

```diff
@@ -1160,10 +1319,23 @@ public String getComment() {
     */
    @Subscribe
    public void onAcquisitionEnded(AcquisitionEndedEvent event) {
-      curStore_ = null;
-      curPipeline_ = null;
-      currentAcquisition_ = null;
-      studio_.events().unregisterForEvents(this);
+      if (event.getStore().equals(curStore_)) {
+         // Restore original Z position and autofocus if applicable.
+         if (isFocusStageAvailable()) {
+            try {
+               core_.setPosition(zStage_, zStart_);
+               if (autofocusMethod_ != null) {
+                  autofocusMethod_.enableContinuousFocus(autofocusOn_);
+               }
+            } catch (Exception e) {
+               studio_.logs().logError(e);
+            }
+         }
+         curStore_ = null;
+         curPipeline_ = null;
+         currentAcquisition_ = null;
+         studio_.events().unregisterForEvents(this);
+      }
    }
```

Comment:
I may be missing something I dont see anywhere where `AcquisitionEndedEvent` is posted right now, except in a try-catch within the `runAcquisition` function. The correct ending behavior should probably be implemented:

Once `acquistion.areEventsFinished()` returns true, no more events will be executed (i.e. no more hardware will be controlled) and it is safe to move hardware back to initial positions. `acquisition.checkForExceptions()` Can also be called after completion, or at any time during acquisition. It will throw an exception if an exception was thrown by any of the other threads that the acquisition engine manages (whether or not this function was called, an exception thrown by an internal acquisition engine thread will cause the acquisition to abort) 

I don't think that the `AcquisitonEndedEvent` currently posted at the end of `runAcquisition(SequenceSettings sequenceSettings)` is currently correct, because, exceptions haven't been checked for to trigger that catch block, nor has the acquisition been given time to shut down

Reply:
The `AcquisitionEndedEvent `is posted in the finally clause of the `AcqEngJDataSink.run()` method.  I added some check before generting the Even at the end of `runAcquisition` (commit 875e24714).